test/system/150-login.bats

```diff
@@ -107,6 +122,65 @@ function setup() {
 # BEGIN actual tests
 # BEGIN primary podman login/push/pull tests
 
+# FIXME: move the two auto-update tests into 255-auto-update.bats once there
+# are helper functions for starting and stopping a registry.
+
+@test "podman-auto-update --authfile"  {
+    # Test the three supported ways of using authfiles with auto updates
+    # 1) Passed via --authfile CLI flag
+    # 2) Passed via the REGISTRY_AUTH_FILE env variable
+    # 3) Via a label at container creation where 1) and 2) will be ignored
+
+    registry=localhost:${PODMAN_LOGIN_REGISTRY_PORT}
+    image_on_local_registry=$registry/name:tag
+    authfile=$PODMAN_TMPDIR/authfile.json
+
+    # First populate the authfile that we can use for the test.
+    run_podman login --authfile=$authfile \
+        --tls-verify=false \
+        --username ${PODMAN_LOGIN_USER} \
+        --password ${PODMAN_LOGIN_PASS} \
+        $registry
+
+    run_podman tag $IMAGE $image_on_local_registry
+    run_podman push --tls-verify=false --creds "${PODMAN_LOGIN_USER}:${PODMAN_LOGIN_PASS}" $image_on_local_registry
+
+    # Generate a systemd service with the "registry" auto-update policy running
+    # "top" inside the image we just pushed to the local registry.
+    generate_autoupdate_service "" registry top "" "" "" $image_on_local_registry
+    ctr=$cname
+    wait_systemd_service_ready container-$ctr.service
+
+    run_podman 125 auto-update
+    is "$output" \
+       ".*Error: checking image updates for container .*: tls: failed to verify certificate: .*"
+
+    run_podman 125 auto-update --tls-verify=false
+    is "$output" \
+       ".*Error: checking image updates for container .*: authentication required"
+
+    # Test 1)
+    run_podman auto-update --authfile=$authfile --tls-verify=false --dry-run --format "{{.Unit}},{{.Image}},{{.Updated}},{{.Policy}}"
+    is "$output" ".*container-$ctr.service,$image_on_local_registry,false,registry.*" "auto-update works with authfile"
```

Comment:
dot-stars (here & lines 168 & 180, beginning and end of string) are unnecessary and could mask warnings or other problems.